packages/x-components/src/x-modules/search/store/__tests__/actions.spec.ts

```diff
@@ -230,6 +232,47 @@ describe('testing search module actions', () => {
     });
   });
 
+  describe('saveSearchResponse', () => {
+    it('saves the search response in the search state', () => {
+      store.dispatch('saveSearchResponse', {
+        ...searchResponseStub,
+        partialResults: [...partialResultsStub]
+      });
+      expect(store.state.results).toEqual(resultsStub);
+      expect(store.state.facets).toEqual(facetsStub);
+      expect(store.state.banners).toEqual(bannersStub);
+      expect(store.state.partialResults).toEqual(partialResultsStub);
+      expect(store.state.promoteds).toEqual(promotedsStub);
+      expect(store.state.redirections).toEqual(redirectionsStub);
+      expect(store.state.spellcheckedQuery).toEqual('');
+      expect(store.state.page).toEqual(1);
+      expect(store.state.config.pageSize).toEqual(24);
+      expect(store.state.queryTagging).toEqual(searchResponseStub.queryTagging);
+    });
+
+    // eslint-disable-next-line max-len
+    it('saves default values of optional or undefined response properties in the search state', () => {
+      store.dispatch('saveSearchResponse', {
+        ...searchResponseStub,
+        partialResults: undefined,
+        redirections: undefined,
+        banners: undefined,
+        promoteds: undefined,
+        spellcheck: undefined
```

Comment:
Looks like the tests are failing because these undefined, adding `as any` should fix them.

Reply:
My bad. I had a `search-adapter` build locally which had a modified `SearchResponse` type version. It should be fine now, as that change got merged into `main` and I've updated the pull request with it.